networkparams_test.go

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2016 The btcsuite developers
+// Copyright (c) 2016 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/decred/dcrd/blockchain"
+	"github.com/decred/dcrd/chaincfg"
+)
+
+//TestDecredNetworkSettings checks Network-specific settings
+func TestDecredNetworkSettings(t *testing.T) {
+
+	checkPowLimitsAreConsistent(t, chaincfg.MainNetParams)
+	checkPowLimitsAreConsistent(t, chaincfg.TestNet2Params)
+	checkPowLimitsAreConsistent(t, chaincfg.SimNetParams)
+
+	checkGenesisBlockRespectsNetworkPowLimit(t, chaincfg.MainNetParams)
+	checkGenesisBlockRespectsNetworkPowLimit(t, chaincfg.TestNet2Params)
+	checkGenesisBlockRespectsNetworkPowLimit(t, chaincfg.SimNetParams)
+
+}
+
+// checkPowLimitsAreConsistent ensures PowLimit and PowLimitBits are consistent with each other
+// PowLimit:         mainPowLimit,// big int
+// PowLimitBits:     0x1d00ffff,  //conceptually the same
+//                                //value, but in an obscure form
+func checkPowLimitsAreConsistent(t *testing.T, params chaincfg.Params) {
+
+	powLimitBigInt := params.PowLimit
+	powLimitCompact := params.PowLimitBits
+
+	toBig := blockchain.CompactToBig(powLimitCompact)
+	toCompact := blockchain.BigToCompact(powLimitBigInt)
+
+	//Check params.PowLimitBits matches params.PowLimit converted into the compact form
```

Comment:
Lacks space after `//` and does not end with a period.  Several other instances of this as well.